src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -241,11 +198,75 @@ public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, Cli
                 // Disposing the handler will not affect any active stream wrapped in the WebSocket.
                 if (disposeHandler)
                 {
-                    handler?.Dispose();
+                    invoker?.Dispose();
                 }
             }
         }
 
+        private static SocketsHttpHandler SetupHandler(ClientWebSocketOptions options, out bool disposeHandler)
+        {
+            SocketsHttpHandler? handler;
+
+            // Create the handler for this request and populate it with all of the options.
+            // Try to use a shared handler rather than creating a new one just for this request, if
+            // the options are compatible.
+            if (options.Credentials == null &&
+                !options.UseDefaultCredentials &&
+                options.Proxy == null &&
+                options.Cookies == null &&
+                options.RemoteCertificateValidationCallback == null &&
+                options._clientCertificates?.Count == 0)
```

Comment:
```suggestion
                (options._clientCertificates?.Count ?? 0) == 0)
```

Assuming we want to match the `certs is null` case here as well.

I see this was the case before your PR, but I don't think this logic was reusing handlers in the common case.